components/suggest/src/db.rs

```diff
@@ -979,4 +990,45 @@ impl<'a> SuggestDao<'a> {
         };
         self.put_meta(UNPARSABLE_RECORDS_META_KEY, unparsable_records)
     }
+
+    /// Stores global Suggest configuration data.
+    pub fn put_global_config(&mut self, config: &SuggestGlobalConfig) -> Result<()> {
+        self.put_meta(GLOBAL_CONFIG_META_KEY, serde_json::to_string(config)?)
+    }
+
+    /// Gets the stored global Suggest configuration data or a default config if
+    /// none is stored.
+    pub fn get_global_config(&self) -> Result<SuggestGlobalConfig> {
+        self.get_meta::<String>(GLOBAL_CONFIG_META_KEY)?
+            .map_or_else(
+                || Ok(SuggestGlobalConfig::default()),
+                |json| Ok(serde_json::from_str(&json)?),
+            )
+    }
+
+    /// Stores configuration data for a given provider.
+    pub fn put_provider_config(
+        &mut self,
+        provider: SuggestionProvider,
+        config: &SuggestProviderConfig,
+    ) -> Result<()> {
+        self.put_meta(
+            &provider_config_meta_key(provider),
+            serde_json::to_string(config)?,
+        )
+    }
+
+    /// Gets the stored configuration data for a given provider or None if none
+    /// is stored.
+    pub fn get_provider_config(
+        &self,
+        provider: SuggestionProvider,
+    ) -> Result<Option<SuggestProviderConfig>> {
+        self.get_meta::<String>(&provider_config_meta_key(provider))?
+            .map_or_else(|| Ok(None), |json| Ok(serde_json::from_str(&json)?))
+    }
+}
+
+fn provider_config_meta_key(provider: SuggestionProvider) -> String {
+    format!("{}{}", PROVIDER_CONFIG_META_KEY_PREFIX, provider as u8)
```

Comment:
Question: since we don't have a `suggestion_provider` table to store this enum in the db. We'd have to reference the source to decode provider config for each key, which might get annoying for debugging. How about we store the string presentation instead in the meta table (i.e. to implement "std::fmt::Display" for `SuggestionProvider`)? Or we could create a table `suggestion_provider` as other tables with the `provider` column can also use it to get the human readable providers.

WDYT?

Reply:
Yeah, I like both ideas. wdyt about continuing to use `PROVIDER_CONFIG_META_KEY_PREFIX` in that case? I'm hesitant to store names or u8's by themselves in a `meta` column since we'd need to be careful to avoid name conflicts with other meta keys. Actually I like your `suggestion_provider` table idea because then it could store both names and config -- a table for provider info.

Reply:
All right, I'll go ahead and merge this now then, thanks.

Reply:
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1878941 so we don't lose track of this idea.